src/Datasource/RuleInvoker.php

```diff
@@ -117,8 +117,20 @@ public function __invoke(EntityInterface $entity, array $scope): bool
     {
         $rule = $this->rule;
         $pass = $rule($entity, $this->options + $scope);
-        if ($pass === true || empty($this->options['errorField'])) {
-            return $pass === true;
+        if ($pass === true) {
+            return true;
+        }
+
+        if (empty($this->options['errorField'])) {
+            // If no errorField is specified but the rule didn't return true,
+            // we should still set a general error on the entity
+            if ($pass !== false && is_string($pass)) {
+                $entity->setError('_rule', [$pass]);
+            } elseif ($pass !== false) {
+                $entity->setError('_rule', ['Application rule failed']);
```

Comment:
Why set a hard coded message instead of using the `$message` below?

I think a much simpler fix would be to modify the line `$errorField = $this->options['errorField'];` below to `$errorField = $this->options['errorField'] ?? '_rule';`

Reply:
You are right, it should

    - First tries $this->options['errorField']
    - Falls back to $this->name if available
    - Finally defaults to '_rule' if neither is set